app/utils.py

```diff
@@ -51,11 +82,39 @@ def mkQueryDist(query):
         # Take first word in vecs_to_add to start addition
         vbase = array([float(i) for i in vecs_to_add[0].vector.split(',')])
         for item in range(1, len(vecs_to_add)):
-            vbase = vbase + array([float(i) for i in vecs_to_add[item].vector.split(',')])
-
+	    w = vecs_to_add[item]
+	    if w in entropies and math.log(entropies[w]+1) > 0:
+		weight=float(1)/float(math.log(entropies[w]+1))
+		vbase = vbase + weight*array([float(i) for i in vecs_to_add[item].vector.split(',')])
+	    else:
+		vbase = vbase + array([float(i) for i in vecs_to_add[item].vector.split(',')])
+
+    vbase=normalise(vbase)
     return vbase
 
 
+##############################################
+# Read pears  ids
+##############################################
+
+def readPears():
+	shared_pears_ids = os.path.join(
+	    os.path.dirname(__file__),
+	    "shared_pears_ids.txt")
```

Comment:
For some reason, this symlink appears to be broken in my codebase.  My python interpreter is not picking up this file from the app/ directory. Investigating.